web/src/main/java/org/springframework/security/web/firewall/StrictHttpFirewall.java

```diff
@@ -297,6 +305,12 @@ public void setAllowUrlEncodedPercent(boolean allowUrlEncodedPercent) {
 		}
 	}
 
+	public void setAllowedHostnames(Predicate<String> allowedHostnames) {
+		if (allowedHostnames != null) {
```

Comment:
Let's please have this throw an exception if a `null` predicate is given. Something like:

```java
if (allowedHostnames == null) {
    throw new IllegalArgumentException("allowedHostnames cannot be null");
}
```